### PR TITLE
Fix Apple Vuln

### PR DESCRIPTION
**Pull request by dfed**
Verified via manual keychain inspection that this fixes the apple vuln. Would prefer to have an automated test. Looking into it.

### CODE REVIEW

**Comment by dfed** [comment]
cc @mikelikespie @EricMuller22 @martinmroz

File: Valet/VALValet.m [hunk 1]
```diff
@@ -428,11 +428,17 @@ - (BOOL)setObject:(NSData *)value forKey:(NSString *)key options:(NSDictionary *
     
     __block OSStatus status = errSecNotAvailable;
     VALExecuteBlockInLock(^{
+#if TARGET_OS_IPHONE
```
Comment (mikelikespie) [inline Valet/VALValet.m]: Kinda difficult to read the control flow with this preprocessor stuff. Either some commments or something else may help.

Comment (dfed) [inline Valet/VALValet.m]: Yeah I thought of that. I might just repeat some code here instead.

**Comment by mikelikespie** [comment]
LGTM

File: Valet/VALValet.m (hunk 1 above)
Comment (EricMuller22) [inline Valet/VALValet.m]: Agreed. Preprocessor conditional + regular conditional makes this harder than it ought to be to reason about.

**Comment by EricMuller22** [comment]
That test 👍🏻

File: Valet/VALValet.m (hunk 1 above)
Comment (dfed) [inline Valet/VALValet.m]: Just made it more legible. take a look

File: Valet/VALSecureEnclaveValet.h [hunk 2]
```diff
@@ -47,4 +47,7 @@ NS_CLASS_AVAILABLE_IOS(8_0)
 /// This method is not supported on VALSecureEnclaveValet.
 - (NSSet *)allKeys __attribute__((unavailable("VALSecureEnclaveValet does not support -allKeys")));
 
+/// This method is not supported on VALSecureEnclaveValet.
+- (BOOL)removeAllObjects __attribute__((unavailable("VALSecureEnclaveValet does not support -allKeys")));
```
Comment (efirestone) [inline Valet/VALSecureEnclaveValet.h]: Copy/paste error on the message.

Comment (dfed) [inline Valet/VALSecureEnclaveValet.h]: great catch. fixed!

**Comment by martinmroz** [comment]
Looks great! :+1:

**Comment by martinmroz** [comment]
GORM

**Comment by dfed** [comment]
cc @emonti

File: Valet/VALValet.m [hunk 3]
```diff
@@ -440,6 +441,18 @@ - (BOOL)setObject:(NSData *)value forKey:(NSString *)key options:(NSDictionary *
             status = VALAtomicSecItemAdd((__bridge CFDictionaryRef)keychainData, NULL);
         }
     }, self.lockForSetAndRemoveOperations);
```
Comment (emonti) [inline Valet/VALValet.m]: This may still be vulnerable to a race condition

File: Valet/VALValet.m [hunk 4]
```diff
@@ -440,6 +441,18 @@ - (BOOL)setObject:(NSData *)value forKey:(NSString *)key options:(NSDictionary *
...
+#else
```
Comment (emonti) [inline Valet/VALValet.m]: This block may still be vulnerable to a race condition

**Comment by emonti** [comment]
LGTM

**Pull request merged**